vlm/src/main/scala/geotrellis/contrib/vlm/package.scala

```diff
@@ -69,6 +75,118 @@ package object vlm {
     new StreamingByteReader(rr)
   }
 
+  implicit class tokenMethods(val token: Long) extends AnyVal {
+
+    def getProjection(): Option[String] = {
+      val crs = Array.ofDim[Byte](1 << 16)
+      GDALWarp.get_crs_wkt(token, 0, crs)
+      Some(new String(crs, "UTF-8"))
+    }
+
+    def rasterExtent(): RasterExtent = {
+      val transform = Array.ofDim[Double](6)
+      val width_height = Array.ofDim[Int](2)
+      GDALWarp.get_transform(token, 0, transform)
+      GDALWarp.get_width_height(token, 0, width_height)
+
+      val x1 = transform(0)
+      val y1 = transform(3)
+      val x2 = x1 + transform(1) * width_height(0)
+      val y2 = y1 + transform(5) * width_height(1)
+      val e = Extent(
+        math.min(x1, x2),
+        math.min(y1, y2),
+        math.max(x1, x2),
+        math.max(y1, y2))
+
+      RasterExtent(e,
+        math.abs(transform(1)), math.abs(transform(5)),
+        width_height(0), width_height(1))
+    }
+
+    def resolutions: List[RasterExtent] = {
+      val N = 1 << 8;
```

Comment:
So we're limited by 256 overviews. Mb we can move this into `Constants`? @jpolchlo had an idea to get rid of unnamed and random values from the GeoTrellis core: https://github.com/locationtech/geotrellis/issues/2872

Reply:
Alot of what you see here is stub code.  I want to test this before concertizing too much.  This limit is ultimately imposed by the number of available file descriptors (other factors) and not by my code.

Reply:
Also, I don't think this limit affects the number of overviews (at least not GDAL overviews), this is to do with datasets.